plugins/imfile/imfile.c

```diff
@@ -567,6 +566,7 @@ openFile(lstn_t *pLstn)
 		if(pLstn->pStrm != NULL)
 			strm.Destruct(&pLstn->pStrm);
 		CHKiRet(strm.Construct(&pLstn->pStrm));
+		CHKiRet(strm.SetbReopenOnTruncate(pLstn->pStrm, pLstn->reopenOnTruncate));
```

Comment:
Have a look at the guarding ```if```: this code is only executed if the state file did not exist. So if it exists, the flag is never applied.